File: 4D4Progr4m5/Knapsack.py
def max(a,b):
    if a>b:
        return a
    else:
        return b
def knapsack(wt,wei,prof,n):
    if n==0 or wt ==0:
        return 0
    if wei[n-1]>wt:
        return knapsack(wt,wei,prof,n-1)
    else:
        return max(prof[n-1] + knapsack(wt-wei[n-1],wei,prof,n-1),knapsack(wt,wei,prof,n-1))

profit = [60,100,120]
weight = [10,20,30]
w =125
n = len(profit)
print("Maximum profit is ") 
print(knapsack(w,weight,profit,n))

File: 4D4Progr4m5/Knapsack.c
# include<stdio.h>

int max(int a, int b)
{
    return (a>b)?a:b;
}

int knapsack(int w, int wei[], int val[], int n)
{
    if(n==0||w==0)
    return 0;

    if(wei[n-1]>w)
    return knapsack(w,wei,val,n-1 );
    else
    return max(val[n-1] + knapsack(w-wei[n-1], wei, val, n-1), knapsack(w, wei, val, n-1));
}

int main()
{
    int weight[]={60,33,100};
    int profit[]={333,44,888}; 
    int w, n;
    w = 125;
    n = sizeof(profit)/sizeof(profit[0]);
    printf("Maximum profit is \n");
    printf("%d\n", knapsack(w,weight,profit,n));
    return 0;
}